### ISSUE TITLE
UI Element List - DetailPress

### ISSUE THREAD

**Issue by thomasmarkert** [oblomov-dev/abap2UI5#208]
Hey,

I have defined the UI Element List in such a way that I trigger the "DetailPress" event. How can I query on which line the event was triggered?

`  
METHOD Z2UI5_IF_APP~MAIN.

    IF check_initialized = abap_false.
      check_initialized = abap_true.

      t_tab = VALUE #(
        ( title = 'Peter'   info = 'Information'    descr = 'this is a description1 1234567890 1234567890'  icon = 'sap-icon://badge'      highlight = 'Information' wrapCharLimit = '100' )
        ( title = 'Peter1'  info = 'Success'        descr = 'this is a description2 1234567890 1234567890'  icon = 'sap-icon://favorite'   highlight = 'Success'  wrapCharLimit = '10')
        ( title = 'Peter2'  info = 'Warning'        descr = 'this is a description3 1234567890 1234567890'  icon = 'sap-icon://employee'   highlight = 'Warning'  wrapCharLimit = '100')
        ( title = 'Peter3'  info = 'Error'          descr = 'this is a description4 1234567890 1234567890'  icon = 'sap-icon://accept'     highlight = 'Error'  wrapCharLimit = '10' )
        ( title = 'Peter4'  info = 'None'           descr = 'this is a description5 1234567890 1234567890'  icon = 'sap-icon://activities' highlight = 'None'  wrapCharLimit = '10')
        ( title = 'Peter5'  info = 'Information'    descr = 'this is a description6 1234567890 1234567890'  icon = 'sap-icon://account'    highlight = 'Information'   wrapCharLimit = '100' )
      ).

    ENDIF.

    CASE client->get( )-event.
      When 'TEST'.
        DATA(lt_sel1) = t_tab.
        client->popup_message_box( `Test` ).

      WHEN 'SELCHANGE'.
        DATA(lt_sel) = t_tab.
        DELETE lt_sel WHERE selected = abap_false.
        client->popup_message_box( `go to details for item ` && lt_sel[ 1 ]-title ).

      WHEN 'BACK'.
        client->nav_app_leave( client->get_app( client->get( )-id_prev_app_stack ) ).
    ENDCASE.

    DATA(page) = z2ui5_cl_xml_view=>factory( )->shell(
        )->page(
            title          = 'abap2UI5 - List'
            navbuttonpress = client->_event( 'BACK' )
              shownavbutton = abap_true
            )->header_content(
                )->link(
                    text = 'Source_Code'  target = '_blank'
                    href = z2ui5_cl_xml_view=>hlp_get_source_code_url( app = me get = client->get( ) )
            )->get_parent( ).

    page->list(
        headertext      = 'List Ouput'
        items           = client->_bind( t_tab )
        mode            = `SingleSelectMaster`
        selectionchange = client->_event( 'SELCHANGE' )
    )->_generic(
         name = `StandardListItem`
            t_prop = VALUE #(
                ( n = `title`       v = '{TITLE}' )
                ( n = `description` v = '{DESCR}' )
                ( n = `icon`        v = '{ICON}' )
                ( n = `iconInset`   v = 'false' )
                ( n = `highlight`   v = '{HIGHLIGHT}' )
                ( n = `info`        v = '{INFO}' )
                ( n = `infoState`   v = '{HIGHLIGHT}' )
*                ( n = `infoStateInverted`   v = 'true' )
                ( n = 'type'      v = `Detail` )
                ( n = 'wrapping'      v = `true` )
                ( n = 'wrapCharLimit'      v = `{WRAPCHARLIMIT}` )
                ( n = 'selected'    v = `{SELECTED}` )
                ( n = 'detailPress'      v = client->_event( 'TEST' ) )

                ) ).

    client->set_next( VALUE #( xml_main = page->get_root(  )->xml_get( ) ) ).

  ENDMETHOD.`

Best Regards 
Thomas

**Comment by choper725** [comment]
{SELECTED} is empty ?

**Comment by oblomov-dev** [comment]
Yes @choper725  is right, check demo Z2UI5_CL_APP_DEMO_03, the  row "SELECTED" is filled:
<img width="1273" alt="image" src="https://user-images.githubusercontent.com/102328295/235350828-c9a14749-564c-41d1-989e-bbacad0512b0.png">
<img width="641" alt="image" src="https://user-images.githubusercontent.com/102328295/235350844-cfe9ef7a-febd-48b4-927c-351b25f8101c.png">

**Comment by thomasmarkert** [comment]
Hey,

No, unfortunately not.
When I click the pen, the "Test" event is triggered, but the "selected" field in the "t_tab" table is not filled.

![image](https://user-images.githubusercontent.com/123086939/235859315-2dfdfcd6-c989-4959-bf15-a046598b4911.png)
![image](https://user-images.githubusercontent.com/123086939/235859087-3eece4a8-59c7-49dc-93b9-143a16a17bcf.png)
![image](https://user-images.githubusercontent.com/123086939/235858872-a45aca4f-a43c-4517-895a-cd302ff43441.png)

Best Regards
Thomas

**Comment by choper725** [comment]
i see, you can try developing a new class for the press of the detailPress event and pass the value of the row,
look in the demos under flow-logic

**Comment by thomasmarkert** [comment]
Can you please tell me the class.

**Comment by choper725** [comment]
z2ui5_cl_app_demo_24

**Comment by thomasmarkert** [comment]
Sorry for my constant questions. I don't know how to do this. With the class mentioned, I can only see how I can transport data from one application to another. Could you please give me the basic code. I hope I can develop the rest myself.

**Comment by choper725** [comment]
on a second tought it wont work with the detailpress but it will work if you change the type = “Navigation” like in @oblomov-dev pic, it will give you an arrow insted of the pencil and youll have values in {SELECTED} which will be the row number

**Comment by oblomov-dev** [comment]
@thomasmarkert its ok to ask, it helps me to make better examples for the demo section ;)

the problem here is that the edit button uses a different event than the selection:
 ( n = 'detailPress'      v = client->_event( 'TEST' ) )
 
for events on item level it is not so easy to figure out the pressed row without additional javascript, i will check this and let you know.

**Comment by oblomov-dev** [comment]
you can check now class: z2ui5_cl_app_demo_48
next time when you have problems, maybe just copy and paste the whole app that i can import it in my repository quickly.

needed to add some extra logic. i think the best way is to do the following:
        ( n = 'detailPress'      v = client->_event( val = 'TEST' data = `${TITLE}` ) )
and then you get at the event:
    data(lv_row_title) = client->get( )-event_data.
    client->popup_message_box( `event: test - ` && lv_row_title ).

you use now the functionality that you can also send generic data with the event method. in this case you import a specific value of the row. Don't forget to use the $ symbol. maybe it is better to use a ID column for identification but this is up to you.

in the end you get this after clicking the pencil:
<img width="500" alt="image" src="https://user-images.githubusercontent.com/102328295/236040097-b687bfac-dd1a-4c39-ac5a-f4c243bad2f3.png">

**Comment by thomasmarkert** [comment]
Where did you get so much knowledge from? Really great. Thanks alot. Next time I will provide you the whole class as code.
I still have the topic with the tables as an open question. You had already made UI elements available for the UI table. It would be best if you could provide me with a small generic example.

**Comment by oblomov-dev** [comment]
you can find an example here https://github.com/oblomov-dev/abap2UI5/issues/216 
and i close this issue.
